src/libpriv/rpmostree-core.c

```diff
@@ -2023,6 +2027,20 @@ rpmostree_context_prepare (RpmOstreeContext *self,
         }
     }
 
+  /* Process excludes */
+  for (char **iter = blacklist_pkgnames; iter && *iter; iter++)
+    {
+      const char *pkgname = *iter;
+      hy_autoquery HyQuery query = hy_query_create (sack);
+      hy_query_filter (query, HY_PKG_NAME, HY_EQ, pkgname);
```

Comment:
Hmm, so one tricky thing about this is that it works on the package name directly, not all provides. And I think that's correct, because AFAICT, `exclude=` works only on the package name too. Though it means that e.g. a `foobar-ng` could slip by even if we had a `package-blacklist: [foobar]`.

Reply:
Right; I think that's basically OK.